server/bundler/loader.js

```diff
@@ -66,6 +66,7 @@ function splitTemplate( path, module, chunkName ) {
 
 	result = [
 		'page( ' + path + ', function( context, next ) {',
+		'	context.sectionRouteMatched = true;',
```

Comment:
This is specific to `stats` section. Due to the catch-all `page` at the end of `my-sites/stats/index`, my 404 route handler won't ever be called, even if you go to `/stats/123/12/4`. [Link to the line](https://github.com/Automattic/wp-calypso/blob/master/client/my-sites/stats/index.js#L47):

Instead, it redirects to stats overview (`/stats`). But while redirecting, it firstly shows an empty `#primary` with sidebar for me, for half a second; screenshot: ![](https://i.imgur.com/z2Hfdcj.png)

It is not clear for me from [this commit](https://github.com/Automattic/calypso-pre-oss/commit/eecff69647aa17d7b12963688c1a238a9f28c4c3) why the last `page` was added. I think we should fix this and show 404 page instead of redirect. But for that, we will need to either remove the last catch-all `page` or to look into the logic of all its route handlers and decide to call `next()` at some point, so that my 404 handler gets the chance to be called.